### PR TITLE
#162727423 Implement create art feature

### PR DESCRIPTION
**Pull request by wombolo**
#### What does this PR do?
Implements create art feature

#### Description of Task to be completed?
- implement artist are able to create artworks and post them
- implement uploading art media files to cloudinary

#### How should this be manually tested?
Login as an artist
Visit `localhost:5050/create-art`
Fill the necessary form fields

#### Any background context you want to provide?
User creating the art must be of type artist

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/162727423

#### Screenshots (if appropriate)
image

#### Questions:

### CODE REVIEW

**Review by abejide001 — APPROVED** [review]
LGTM

File: src/actions/artActions.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  let imageURLs = [];
+  dispatch(setUploadSuccess());
+
+  const {files: images} = payload;
+
+  const uploads = await images.map(async (image)=> {
+    const formData = new FormData();
+    formData.append("file", image);
+    formData.append("upload_preset", "artcave_articles"); // Replace the preset name with your own
+    formData.append("api_key", "199196371633358"); // Replace API key with your own Cloudinary API key
```
Comment (andela-dbamidele) [inline src/actions/artActions.js]: Please store API key in env.

File: src/actions/artActions.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      `${basePath}/arts/`,
+      {
+        ...payload,
+        media: JSON.stringify(mediaFiles[0])
+      },
+      { headers: {
+          "Content-Type": "application/json",
+          "x-access-token": payload.accessToken
+        }});
+    dispatch(window.location.replace(`/arts/${postToBackEnd.data.data.slugifiedTitle}`));
```
Comment (andela-dbamidele) [inline src/actions/artActions.js]: Why are you calling dispatch here?

File: src/components/Logo.jsx [hunk 3]
```diff
@@ -14,7 +14,7 @@ export default class Logo extends Component {
       <NavLink
       className="navbar-brand col-lg-2 logo pull-left w-50 p-0"
       to="/" id="brand-logo">
-        <img alt={'logo'} src='../assets/images/artcave-logo.png' className='w-25' />artCave</NavLink>
+        <img alt={'logo'} src={require('../assets/images/artcave-logo.png')} className='w-25' />artCave</NavLink>
```
Comment (andela-dbamidele) [inline src/components/Logo.jsx]: Please remove inline `require`

File: src/components/Menu.jsx [hunk 4]
```diff
@@ -30,27 +31,24 @@ export class Menu extends Component {
...
+          <li className="nav-item dropdown">
+            <a className="nav-link d-inline-flex text-capitalize" href="#" id="navbarDropdown"
+               role="button" data-toggle="dropdown" aria-haspopup="true"
+               aria-expanded="false"> {user.registeredUsername || user.username}<i className='fa fa-caret-down m-2'/>
+            </a>
+            <div className="dropdown-menu" aria-labelledby="navbarDropdown">
+              <NavLink to="/profile" className="dropdown-item">My Profile</NavLink>
+              <NavLink to="/create-art" className={`dropdown-item ${userType === 'artist' ? '' : 'd-none'}`}>Create Art</NavLink>
+              <div className="dropdown-divider"/>
+              <NavLink to="/login" onClick={this.handleLogout} id="nav-link" className="dropdown-item">logout</NavLink>
```
Comment (andela-dbamidele) [inline src/components/Menu.jsx]: Are you sure you want to log the user out when you click on the login link?

**Review by andela-dbamidele — CHANGES_REQUESTED** [review]
Minor changes needed. Nice work though.

File: src/actions/artActions.js (hunk 2 above)
Comment (wombolo) [reply src/actions/artActions.js]: It was an error, I'd remove it

File: src/actions/artActions.js (hunk 1 above)
Comment (wombolo) [reply src/actions/artActions.js]: Okay, will do

File: src/components/Logo.jsx (hunk 3 above)
Comment (wombolo) [reply src/components/Logo.jsx]: Okay, I'll remove it

File: src/components/Menu.jsx (hunk 4 above)
Comment (wombolo) [reply src/components/Menu.jsx]: I'll make the corrections. Thanks

**Pull request merged**